src/System.EventLog/src/System/Diagnostics/EventLogInternal.cs

```diff
@@ -0,0 +1,1550 @@
+
+                string msg = null;
+                try
+                {
+                    msg = EventLog.TryFormatMessage(hModule, messageNum, insertionStrings);
+                }
+                finally
+                {
+                    if (!IsOpen)
+                    {
+                        hModule.Close();
+                    }
+                }
+
+                if (msg != null)
+                {
+                    return msg;
+                }
+
+            }
+            return null;
+        }
+
+        internal EventLogEntry[] GetAllEntries()
+        {
+            // we could just call getEntryAt() on all the entries, but it'll be faster
+            // if we grab multiple entries at once.
+            string currentMachineName = this.machineName;
+
+            if (!IsOpenForRead)
+                OpenForRead(currentMachineName);
+
+            EventLogEntry[] entries = new EventLogEntry[EntryCount];
+            int idx = 0;
+            int oldestEntry = OldestEntryNumber;
+
+            int bytesRead;
+            int minBytesNeeded;
+            int error = 0;
+            while (idx < entries.Length)
+            {
+                byte[] buf = new byte[BUF_SIZE];
+                bool success = UnsafeNativeMethods.ReadEventLog(readHandle, NativeMethods.FORWARDS_READ | NativeMethods.SEEK_READ,
+                                                      oldestEntry + idx, buf, buf.Length, out bytesRead, out minBytesNeeded);
+                if (!success)
+                {
+                    error = Marshal.GetLastWin32Error();
+                    Debug.WriteLineIf(CompModSwitches.EventLog.TraceVerbose, "Error from ReadEventLog is " + error.ToString(CultureInfo.InvariantCulture));
+#if !RETRY_ON_ALL_ERRORS
+                    if (error == NativeMethods.ERROR_INSUFFICIENT_BUFFER || error == NativeMethods.ERROR_EVENTLOG_FILE_CHANGED)
+                    {
+#endif
+                        if (error == NativeMethods.ERROR_EVENTLOG_FILE_CHANGED)
+                        {
+                            Reset(currentMachineName);
+                        }
+                        // try again with a bigger buffer if necessary
+                        else if (minBytesNeeded > buf.Length)
+                        {
+                            Debug.WriteLineIf(CompModSwitches.EventLog.TraceVerbose, "Increasing buffer size from " + buf.Length.ToString(CultureInfo.InvariantCulture) + " to " + minBytesNeeded.ToString(CultureInfo.InvariantCulture) + " bytes");
+                            buf = new byte[minBytesNeeded];
+                        }
+                        success = UnsafeNativeMethods.ReadEventLog(readHandle, NativeMethods.FORWARDS_READ | NativeMethods.SEEK_READ,
+                                                         oldestEntry + idx, buf, buf.Length, out bytesRead, out minBytesNeeded);
+                        if (!success)
+                            break;
+#if !RETRY_ON_ALL_ERRORS
```

Comment:
same again - RETRY_ON_ALL_ERRORS is never defined, leave the contents

Reply:
done